x-pack/plugins/profiling/server/index.ts

```diff
@@ -9,8 +9,19 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import type { PluginConfigDescriptor, PluginInitializerContext } from '@kbn/core/server';
 import { ProfilingPlugin } from './plugin';
 
+const packageInputSchema = schema.object({
+  host: schema.string(),
+  secret_token: schema.maybe(schema.string()),
+  tls_enabled: schema.boolean(),
+  tls_supported_protocols: schema.arrayOf(schema.string()),
+  tls_certificate_path: schema.string(),
+  tls_key_path: schema.string(),
```

Comment:
Same here, these keys could be non existing in a config when `tls_enabled` is == false, but the config would still be valid

Reply:
@inge4pres / @florianl I made all props optional. There is one thing though. Event though we don't specify any configuration, for the `collector` we will always have the `secret_token` automatically set. Is it ok?